v2/src/log_stacktrace.c

```diff
@@ -0,0 +1,279 @@
+                    wasContextAcquired = true;
+                }
+                wasThreadSuspended = true;
+            }
+        }
+
+        if (!wasContextAcquired)
+        {
+            /*not much to do without the thread context*/
+        }
+        else
+        {
+            /*all following function calls are protected by the same SRW*/
+            AcquireSRWLockExclusive(&lockOverSymCalls);
+            {
+                STACKFRAME64 stackFrame = { 0 };
+                stackFrame.AddrPC.Offset = context.INSTRUCTION_POINTER_REGISTER;
+                stackFrame.AddrPC.Mode = AddrModeFlat;
+                stackFrame.AddrFrame.Offset = context.FRAME_POINTER_REGISTER;
+                stackFrame.AddrFrame.Mode = AddrModeFlat;
+                stackFrame.AddrStack.Offset = context.STACK_POINTER_REGISTER;
+                stackFrame.AddrStack.Mode = AddrModeFlat;
+
+                bool thisIsFirstFrame = true; /*used to skip the first frame, which is "us", that is don't display information about "getThreadStackAsString"*/
+
+                /*4) once the context has been acquired, call StackWalk64 to get the stack frames. For every frame:*/
+                while (StackWalk64(
+                    IMAGE_FILE_MACHINE_AMD64,
+                    hProcess,
+                    hThread,
+                    &stackFrame,
+                    &context,
+                    NULL,
+                    SymFunctionTableAccess64,
+                    SymGetModuleBase64,
+                    NULL))
+                {
+
+                    if (thisIsFirstFrame)
+                    {
+                        thisIsFirstFrame = false; /*no printing for the top of the stack, which is "us". us = getThreadStackAsString*/
+                        continue;
+                    }
+
+                    DWORD64 displacement = 0;
+
+                    SYMBOL_INFO_EXTENDED buffer;
+
+                    PSYMBOL_INFO pSymbol = &buffer.symbol;
+
+                    pSymbol->SizeOfStruct = sizeof(SYMBOL_INFO);
+                    pSymbol->MaxNameLen = MAX_SYM_NAME;
+
+                    const char* function_name;
+
+                    /*4.1) determine the function name*/
+                    if (!SymFromAddr(hProcess, stackFrame.AddrPC.Offset, &displacement, pSymbol))
+                    {
+                        function_name = "failure in SymFromAddr";
+                    }
+                    else
+                    {
+                        function_name = pSymbol->Name;
+                    }
+
+                    IMAGEHLP_LINE64 line;
+                    line.SizeOfStruct = sizeof(IMAGEHLP_LINE64);
+                    DWORD lineDisplacement;
+                    const char* file_name;
+                    uint32_t line_number;
+
+                    /*4.2) determine the file name and line number*/
+                    if (!SymGetLineFromAddr64(hProcess, stackFrame.AddrPC.Offset, &lineDisplacement, &line))
+                    {
```

Comment:
Would be nice to have UTs also for this. Because likely the int tests and not injecting any kind of errors.
Likely a separate task.

Reply:
Myeah, another task.

[https://msazure.visualstudio.com/One/_workitems/edit/25732458](https://msazure.visualstudio.com/One/_workitems/edit/25732458)